src/js/Controllers/UIController.js

```diff
@@ -193,6 +200,48 @@ class UIController {
                     this.onSystemContext("HMI_OBSCURED", context)
                 }
 
+                return null
+            case "SubtleAlert":
+
+                var tryAgainInfo = undefined;
+                var activeInteractionId = undefined;
+                const state3 = store.getState()
+                for (var appId in state3.ui) {
+                    var app = state3.ui[appId];
+                    if (app.isPerformingInteraction) {
+                        tryAgainInfo = 'A PerformInteraction is active';
+                        activeInteractionId = app.interactionId;
+                    } else if (app.alert.showAlert) {
+                        tryAgainInfo = (app.alert.subtle ? 'Another SubtleAlert' : 'An Alert') + ' is active';
+                        activeInteractionId = app.alert.msgID;
+                    }
+                }
+
+                if (activeInteractionId) {
+                    var tryAgainTime = this.endTimes[activeInteractionId] - Date.now();
+                    var rpc = RpcFactory.SubtleAlertErrorResponse(rpc.id, 4, tryAgainInfo);
+                    rpc.error.data.tryAgainTime = tryAgainTime;
```

Comment:
i could but it would require an if check that would fail for 2/3 of the calls to `SubtleAlertErrorResponse` and have this exact same line inside just for this one call, in a compiled language I would do the optional param every time but in an interpreted I thought it will cost us more time than the one off case's complexity is worth.

Reply:
I guess I understand the reasoning and for now since there is only one argument so there shouldn't be too much of a problem. However, if we were to add additional optional parameters in the future it would make sense to add the params into the factory function. Since JS doesn't allow function overloading, checking if the optional parameters are undefined(as we do in other RPCFactory functions) would be the only way to go.